.github/workflows/build.yml

```diff
@@ -12,16 +12,20 @@ jobs:
   tests:
     name: Tests
     uses: ./.github/workflows/test.yml
+    permissions:
+      contents: read
 
   generate-build-id:
     name: "Generate Build Id"
     needs: [ tests ]
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
     outputs:
       build-id: ${{ steps.generate.outputs.buildId }}
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
does not actions/checkout@v4 take the latest version? I am just trying to understand why do we need to pull "checkout@11bd71901bbe5b1630ceea73d27597364c9af683" ? looks a bit scary :) ....